src/ProductList.js

```diff
@@ -0,0 +1,86 @@
+import { useState } from "react";
+import styled from "styled-components";
+import Grid from "./Grid";
+
+const Sidebar = styled.div`
+  height: 100%;
+  width: 200px;
+  position: sticky;
+  top: 0px;
+  float: left;
+  background-color: lightblue;
+  padding: 10px 0 20px 0;
+`;
+
+const Content = styled.div`
+  margin-left: 200px;
+  padding: 1px 16px;
+`;
+
+const FilterButton = styled.button`
+  width: 90%;
+  margin: 5px;
+  background-color: ${(props) =>
+    props.active ? "rgb(105, 176, 205)" : "rgb(135, 206, 235)"};
+
+  &:hover {
+    background-color: rgb(255, 255, 255, 0.3);
+  }
+`;
+
+const FilterTittle = styled.div`
+  font-weight: bold;
+  margin: 0px 0 10px 0;
+`;
+
+const filterByCategory = (setActiveFilters, activeFilters, id) => {
+  if (activeFilters.includes(id)) {
+    setActiveFilters(
+      activeFilters.filter((activeFilter) => activeFilter !== id)
+    );
+    console.log(`removed category id:${id}`);
+  } else {
+    setActiveFilters([id, ...activeFilters]);
+    console.log(`added category id:${id}`);
+  }
+};
```

Comment:
Deberíamos mover esta función a un archivo `utils.js` esto implicaría que no podemos usar el setActiveFilters en ella al moverla, por lo tanto debería ser una función puramente de JS. Yo la refactorizaría a algo así para poder utilizarla como import en este componente y en otros de ser necesario: 
```suggestion
// en src/utils/utils.js
export const filterByCategory = (activeFilters, id) => {
  if (activeFilters.includes(id)) {
    return activeFilters.filter((activeFilter) => activeFilter !== id);
  } else {
    return [id, ...activeFilters]);
  }
};
```

Para después utilizarla en este componente así:

```
import { filterByCategory } from './utils/utils.js'

const handleFilterOnClick = (activeFilters, id) => {
  setActiveFilters( filterByCategory(activeFilters, id) )
}

<FilterButton
  onClick={ () =>  handleFilterOnClick(activeFilters, category.id) }
  active={ activeFilters.includes(category.id) }
>
```

Reply:
También estaría bien si pudieras renombrar la función de `filterByCategory` por algo como `getCategoriesToFilterBy` porque tal cual esta función no es la que filtra los productos.